File: app/src/com/ternaryop/tumblr/Callback.java
package com.ternaryop.tumblr;

public interface Callback<T> {
	/**
	 * Used to communicate with tumblr after APIs calls, is always called on main thread
	 * @param tumblr
	 * @param result some API can return data
	 */
	public void complete(T result);
	
	/**
	 * Called when some error stop execution, is always called on main thread
	 * @param tumblr
	 * @param e
	 */
	public void failure(Exception e);
}

File: app/src/main/java/com/ternaryop/tumblr/Callback.kt
package com.ternaryop.tumblr

interface Callback<in T> {
    /**
     * Used to communicate with tumblr after APIs calls, is always called on main thread
     * @param result some API can return data
     */
    fun complete(result: T)

    /**
     * Called when some error stop execution, is always called on main thread
     * @param e the exception that caused the failure
     */
    fun failure(e: Exception)
}
